### PR TITLE
[RLlib] Discontinue support for "hybrid" API stack (using RLModule + Learner, but still on RolloutWorker and Policy)

### PR DESCRIPTION
**Pull request by sven1977**
Discontinue support for "hybrid" API stack (using RLModule + Learner, but still on RolloutWorker and Policy)
* The config setting combination of `enable_rl_module_and_learner=True` AND `enable_env_runner_and_connector_v2=False` is no longer allowed and will create an error.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm_config.py [hunk 1]
```diff
@@ -4200,6 +4200,16 @@ def _validate_new_api_stack_settings(self):
             # `enable_rl_module_and_learner=True`.
             return
 
+        # Disabled hybrid API stack. Now, both `enable_rl_module_and_learner` and
+        # `enable_env_runner_and_connector_v2` must be True or both False.
+        if not self.enable_env_runner_and_connector_v2:
+            raise ValueError(
```
Comment (simonsays1980) [inline rllib/algorithms/algorithm_config.py:4206]: I smell a new world rising .... :)

File: rllib/algorithms/bc/tests/test_bc.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def test_bc_compilation_and_learning_from_offline_file(self):
         min_return_to_reach = 75.0
 
         # Test for RLModule API and ModelV2.
-        for rl_modules in [True, False]:
```
Comment (simonsays1980) [inline rllib/algorithms/bc/tests/test_bc.py:54]: Yessss

File: rllib/policy/policy.py [hunk 3]
```diff
@@ -1389,17 +1344,13 @@ def _initialize_loss_from_dummy_batch(
...
-
+        explore = False
         actions, state_outs, extra_outs = self.compute_actions_from_input_dict(
             self._dummy_batch, explore=explore
         )
-        if not self.config.get("enable_rl_module_and_learner", False):
-            for key, view_req in self.view_requirements.items():
-                if key not in self._dummy_batch.accessed_keys:
-                    view_req.used_for_compute_actions = False
+        for key, view_req in self.view_requirements.items():
```
Comment (simonsays1980) [inline rllib/policy/policy.py:1351]: We should be able to remove now also the `view_requirements` from the `Catalog`, shouldn't we? In addition remove all the `view_requirement`  methods in the RLModule

**Review by simonsays1980 — APPROVED** [review]
LGTM. Please check, if we can remove in the same breath the `view_requirement` from `core.models.Catalog`  and the `view_requirements`-related methods from `RLModule`. In addition: Can't we with this change also deprecate TensorFlow in the new stack?

File: rllib/policy/policy.py (hunk 3 above)
Comment (sven1977) [reply rllib/policy/policy.py:1349]: done

**Pull request merged**